client/driver/docker.go

```diff
@@ -76,6 +76,22 @@ func (d *DockerDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool
 func createHostConfig(task *structs.Task) *docker.HostConfig {
 	// hostConfig holds options for the docker container that are unique to this
 	// machine, such as resource limits and port mappings
+
+        mode, ok := task.Config["network_mode"]
+        if !ok || mode == "" {
+                // docker default
+                logger.Printf("[WARN] driver.docker: no mode specified for networking, defaulting to bridge")
```

Comment:
"defaulting to bridge" should probably change to "using Docker default" or something, since we're no longer actually specifying `bridge` here